src/lib/urls.js

```diff
@@ -30,3 +30,39 @@ export function getCanonicalPath(path) {
   }
   return parts.join('/');
 }
+
+export function copyLinkToClipboard() {
+  if (!('clipboard' in navigator)) {
+    return;
+  }
+  document
+    .querySelector('main')
```

Comment:
Instead of adding a listener to each heading, what do you think about adding a single listener to the body and then doing:

```js
const link = e.target.closest('.w-headline-link');
if (!link) {
  return;
}
```

`.w-headline-link` is the link that we use on the hash anchor. I think I'd rather consolidate the functionality on that element since technically a heading is non-interactive and not clickable.

Reply:
The advantage of my logic is that you can click the headline or the '#' with the same effect. You could get the same with just one global listener, but then the code to disregard irrelevant clicks and to find the closest headline or headline link to show the toast would add complexity. Per article there aren't that many headlines that the current approach would cause a serious issue.